### ISSUE TITLE
Can bao also verify content of unknown length?

### ISSUE THREAD

**Issue by markg85** [oconnor663/bao#53]
Hi,

I get the idea behind boa verified slices but this already assumes the whole data in some shape or form is already somewhere. There is a beginning and an end, just the one getting the data isn't aware of it yet.

My case is when there is no known end at all.
Think of live streaming - as in video - as an example. You don't know when the stream is going to end so from a tree point of view the tree is continuously being changed till the very root.

Besides a live video stream, logfile appending and rotation would be applicable here too. It too would be a file that keeps changing till a certain point (say till a day of logging) after which rotation kicks in and a new file is created.

Does boa have a solution here where you can do verified streaming of such data?

**Comment by oconnor663** [comment]
Yeah that's an interesting question. I'm pretty sure the answer is no, because the design of Bao assumes that the recipient has the root hash of whatever they're trying to verify, which they somehow (outside the scope of Bao itself) know to be correct. If you're live-streaming something, then no one can know the hash in advance. You could compute "snapshot" hashes of the stream as you go, but you still need some way to authenticate those hashes (i.e. a TLS connection to a trusted server), and at that point you could probably just use the same authentication mechanism for the whole stream.
